File: UCF HSPT Documents/2010/Solutions/wakkienunu.cpp
//Include the standard i/o stuff
#include <iostream>
#include <fstream>
using namespace std;
//The alternative to the namespace is
// just calling cin and cout as
// std::cin and std::cout


int main(void)
{
    //The number of appliances, the number of checks,
    // and the base cost, respectively
    ifstream myfile;
    int n, m, cost;

    // Open the input file
    myfile.open("wakkienunu.in");

    myfile >> n;

    // A standard for loop from 0 to n-1 (a total of n times)
    for(int i = 0; i < n; i++)
    {
        //get the name of the appliance
        string app;

        myfile >> app;

        //Get the cost of the appliance
        myfile >> cost;

        //Get m, the number of checks to make
        myfile >> m;

        //Print out the header required by the problem
        cout << "Appliance " << app << ":\n";

        //Now loop over the m checks
        for(int j = 0; j < m; j++)
        {
            //Get the retail vale from input
            int retail;
            myfile >> retail;

            //If we paid more than the base cost
            if(cost < retail)
                cout << "You paid too much!" << endl;
            else if(cost > retail) //else the retail was less than the base cost
                cout << "I love appliances!" << endl;
            else //If it's not greater, and not less, it's equal!
                cout << "Wakkie Nu Nu!" << endl;
            //I have no idea what Wakkie Nu Nu means
        }
        //Print out what extra blank line.
        cout << endl;
    }
    myfile.close();
    return 0;
}

File: UCF HSPT Team/src/y2010/wakkienunu.java
import java.util.*;
import java.io.*;

public class wakkienunu {

	public wakkienunu(Scanner in)
	{
		//First we read in the number of appliances to be checked
		int n = in.nextInt();
		
		// A standard for loop from 0 to n-1 (a total of n times)
		for(int i = 0; i < n; i++)
		{
			//Read in the name of the appliance
			String app = in.next();
			
			//Get the cost of the appliance
			int cost = in.nextInt();
			
			//Get m, the number of checks to make
			int m = in.nextInt();
			
			//Print out the header required by the problem
			System.out.println("Appliance " + app + ":");
			
			//Now loop over the m checks
			for(int j = 0; j < m; j++)
			{
				//Get the retail vale from input
				int retail = in.nextInt();
				
				//If we paid more than the base cost
				if(cost < retail)
					System.out.println("You paid too much!"); 
				else if(cost > retail) //else the retail was less than the base cost
					System.out.println("I love appliances!");
				else //If it's not greater, and not less, it's equal!
					System.out.println("Wakkie Nu Nu!");
				//I have no idea what Wakkie Nu Nu means
			}
			//Print out what extra blank line.
			System.out.println();
		}
	}
	
	//I prefer to create an instance of my sourcecode
	// and do all my logic in the constructor.
	public static void main(String[] args) throws Exception
	{
		new wakkienunu(new Scanner(new File("wakkienunu.in")));
	}
}
